### ISSUE TITLE
ValueError("bg_num_rois = 0 and fg_num_rois = 0, this should not happen!")

### ISSUE THREAD

**Issue by twosatnams** [jwyang/faster-rcnn.pytorch#330]
Hi, I'm training on my own dataset of about 530k photos, all labeled. Most of the photos come from the [DeepFashion](http://mmlab.ie.cuhk.edu.hk/projects/DeepFashion.html) dataset. I'm using VGG as my net on 4 GPUs with a batch size of 32. Approximately 10% into the first epoch I see the following exception raised

`ValueError("bg_num_rois = 0 and fg_num_rois = 0, this should not happen!")`

It always happens while processing a different/unique batch of photos so I'm ruling out the possibility that there is something wrong with the `.jpg` files.

I edited `lib/model/rpn/proposal_target_layer_cascade.py`and logged the values for `bg_num_rois` and `fg_num_rois` for the entire training. Noticed the a trend that `bg_num_rois` stays around 1600-2000 range for the entire training and `fg_num_rois` hovers around 0-20. But `bg_num_rois` suddenly drops to 0 a few batches (1-2) before the crash. When this happens the value for `fg_num_rois` drops to 1 and then finally 0, which results in the exception.

Anybody else facing the same issue or knows how to resolve this?

Thanks

**Comment by mehrdadazizi72** [comment]
I have same issue too

**Comment by twosatnams** [comment]
I think I located the cause, the weights of the RPN model turn `nan` first before running into this error. Reducing the learning rate has helped, but I'm not sure if that is the right solution here

**Comment by Youggie** [comment]
Can you tell us how to sovle this problem ?Reducing the learing rate ?

**Comment by harrywang725** [comment]
Have you checked your gt_boxes.shape? May be there is no gt_boxes,and all overlaps returns 0 .
So no fg ,no bg
